### PR TITLE
Bug 1833886 - Update guide to merging contributor PRs doc

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1833886

### CODE REVIEW

**Review by MatthewTighe — APPROVED** [review]
LGTM, thanks for doing this 🙏

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (MatthewTighe) [reply fenix/docs/Guide-to-merging-contributor-PRs.md:18]: On [this patch](https://github.com/mozilla-mobile/firefox-android/pull/1727) I have only been able to get it to work without the `@`, actually.

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (gabrielluong) [inline fenix/docs/Guide-to-merging-contributor-PRs.md:8]: Perhaps we could just simplify this to be just 
gh pr checkout <PR number>

I have only ran the command (without the fetch) that I see from the following:
Screen Shot 2023-05-18 at 1 04 39 PM

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (gabrielluong) [inline fenix/docs/Guide-to-merging-contributor-PRs.md:20]:
```suggestion
5. Once the UI tests have all completed and passed, add `needs landing` label the contributor's PR.
```
We won't need to manually remove labels after my GitHub actions land.

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (gabrielluong) [inline fenix/docs/Guide-to-merging-contributor-PRs.md:1]: Perhaps also add instructions on how to edit and push changes for contributor PR as well?

**Review by gabrielluong — COMMENTED** [review]
No need to wait for further approval from me.

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (rocketsroger) [reply fenix/docs/Guide-to-merging-contributor-PRs.md:8]: Sure, I'll update. thanks

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (rocketsroger) [reply fenix/docs/Guide-to-merging-contributor-PRs.md:18]: I will try "@bors try" on that patch to see if that works as well

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (rocketsroger) [reply fenix/docs/Guide-to-merging-contributor-PRs.md:1]: Sure, I'll do this as a follow up.  It'll be more complicated.

File: fenix/docs/Guide-to-merging-contributor-PRs.md
Comment (rocketsroger) [reply fenix/docs/Guide-to-merging-contributor-PRs.md]: Sure, I'll remove

**Pull request merged**
